plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/DebuggerPresenter.java

```diff
@@ -139,19 +147,21 @@ public String getTitleToolTip() {
   public void go(AcceptsOneWidget container) {
     container.setWidget(view);
     debuggerToolbar.go(view.getDebuggerToolbarPanel());
+    watchToolbar.go(view.getDebuggerWatchToolbarPanel());
   }
 
   @Override
-  public void onExpandVariablesTree(MutableVariable variable) {
+  public void onExpandVariable(Variable variable) {
     Debugger debugger = debuggerManager.getActiveDebugger();
     if (debugger != null && debugger.isSuspended()) {
       Promise<? extends SimpleValue> promise =
           debugger.getValue(variable, view.getSelectedThreadId(), view.getSelectedFrameIndex());
-
       promise
           .then(
               value -> {
-                view.setVariableValue(variable, value);
+                MutableVariable updatedVariable = new MutableVariableImpl(variable);
```

Comment:
I would check if `variable` is mutable already.

Reply:
What do you mean?

Reply:
It might be already mutable. If so, no need to wrap into another mutable variable

Reply:
OK